mllib/src/main/scala/org/apache/spark/ml/param/sharedParams.scala

```diff
@@ -34,6 +34,43 @@ private[ml] trait HasRegParam extends Params {
   def getRegParam: Double = get(regParam)
 }
 
+private[ml] trait HasElasticNetParam extends HasRegParam {
```

Comment:
Definitely. It seems that there is no easy way to specify the valid parameter range or requirement in this framework. Do you think it's a good idea that I add the check like the following?

```scala
  def getElasticNetParam: Double = {
    val elasticNetParam = get(elasticNetParam)
    require(elasticNetParam >= 0)
    require(elasticNetParam <= 1)
    elasticNetParam
  }
```